Alpaca.Markets/RestClient.cs

```diff
@@ -122,19 +152,46 @@ public void Dispose()
             for(var attempts = 0; attempts < throttler.MaxAttempts; ++attempts)
             {
                 throttler.WaitToProceed();
-
+                HttpStatusCode statusCode = HttpStatusCode.OK;
                 try
                 {
-                    using (var stream = await httpClient.GetStreamAsync(endpointUri))
-                    using (var reader = new JsonTextReader(new StreamReader(stream)))
+                    using (HttpResponseMessage response = await httpClient.GetAsync(endpointUri, HttpCompletionOption.ResponseHeadersRead))
                     {
-                        var serializer = new JsonSerializer();
-                        return serializer.Deserialize<TJson>(reader);
+                        // Adhere to server reported instructions
+                        statusCode = response.StatusCode;
+                        if (statusCode != HttpStatusCode.OK)
+                        {
+                            if (_retryHttpStatuses.Contains((int)statusCode))
+                            {
+                                // Accomodate Retry-After header
+                                var retryAfter = response.Headers.RetryAfter?.Delta?.TotalMilliseconds ?? (response.Headers.RetryAfter?.Date?.LocalDateTime - DateTime.Now)?.TotalMilliseconds;
+                                if (retryAfter != null)
+                                {
+                                    throttler.AllStop((Int32)retryAfter);
+                                }
+                                // Server unavailable, or Too many requests (429 can happen when this client competes with another client, e.g. mobile app)
+                                else if (statusCode == (HttpStatusCode)429 || statusCode == (HttpStatusCode)503)
+                                {
+                                    // TODO: If server logic fixed to provide Retry-After, this will be dead code to remove
+                                    throttler.AllStop(3000);
```

Comment:
Curious about your opinion of this.  When the server doesn't say how long to wait, I couldn't think of anything better to do than some arbitrary wait period.

Reply:
My suggestion - use randomly generated delay (from 1 to 5 second).

Reply:
Now randomly generates delay when server fails to specify